ibc-core/ics04-channel/types/src/msgs/chan_open_init.rs

```diff
@@ -43,17 +43,30 @@ impl MsgChannelOpenInit {
 impl Protobuf<RawMsgChannelOpenInit> for MsgChannelOpenInit {}
 
 impl TryFrom<RawMsgChannelOpenInit> for MsgChannelOpenInit {
-    type Error = ChannelError;
+    type Error = DecodingError;
 
     fn try_from(raw_msg: RawMsgChannelOpenInit) -> Result<Self, Self::Error> {
         let chan_end_on_a: ChannelEnd = raw_msg
             .channel
-            .ok_or(DecodingError::MissingRawData {
-                description: "channel end not set".to_string(),
-            })?
+            .ok_or(DecodingError::missing_raw_data("channel end"))?
             .try_into()?;
-        chan_end_on_a.verify_state_matches(&State::Init)?;
-        chan_end_on_a.counterparty().verify_empty_channel_id()?;
+        chan_end_on_a
+            .verify_state_matches(&State::Init)
+            .map_err(|_| {
+                DecodingError::invalid_raw_data(format!(
+                    "expected channel end to be in `Init` state but it is in `{}` instead",
+                    chan_end_on_a.state
+                ))
+            })?;
+        chan_end_on_a
+            .counterparty()
+            .verify_empty_channel_id()
+            .map_err(|_| {
```

Comment:
I suggest we call `.channel_id()` here instead and remove `.verify_empty_channel_id()` along with its related error variant under `ChannelError`, since it's not used anywhere else.